tasks/main.yml

```diff
@@ -248,3 +248,8 @@
       - result_accountkey.key_is_consistent
       - result_csr.signature_valid
     msg: "The private keys and CSR are valid"
+
+- name: "Revert python interpreter"
+  set_fact:
+    ansible_python_interpreter: "{{ ansible_facts['discovered_interpreter_python'] }}"
+  when: revert_python_interpreter is defined and revert_python_interpreter == true
```

Comment:
How about changing the `when` condition to `when: ansible_os_family in ['RedHat', 'Darwin']` to match the `when` condition for when the virtualenv tasks are executed? That way, we don't need to use a custom fact.